hoprd/rest-api/src/messages.rs

```diff
@@ -300,6 +335,15 @@ async fn handle_send_message(input: &str, state: Arc<InternalState>) -> Result<(
         Ok(msg) => {
             let hopr = state.hopr.clone();
 
+            let destination = msg.destination.clone().fulfill(hopr.peer_resolver()).await;
```

Comment:
Here you do it better, cloning only `destination` instead of the entire `args`. 

However, it still would be better if `fulfill` was `fn fulfill(self, resolver: ...) -> Self` as no cloning would take place then at all.